R/nrow_subset_linter.R

```diff
@@ -0,0 +1,43 @@
+#' Block usage of `nrow(subset(x, .))`
+#'
+#' Using `nrow(subset(x, condition))` to count the instances where `condition`
+#'   applies inefficiently requires doing a full subset of `x` just to
+#'   count the number of rows in the resulting subset.
+#' There are a number of equivalent expressions that don't require the full
+#'   subset, e.g. `with(x, sum(condition))` (or, more generically,
+#'   `with(x, sum(condition, na.rm = TRUE))`).
+#' The same can be said of other versions of this like
```

Comment:
We should be clear that this is currently not detected.

Reply:
I think I wrote that with the intention of having it implemented but didn't follow through. Dropping the reference for now.